test/secmemtest.c

```diff
@@ -75,12 +75,31 @@ static int test_sec_mem(void)
     TEST_ptr_null(OPENSSL_secure_malloc((size_t)-1));
     TEST_true(CRYPTO_secure_malloc_done());
 
+    /*-
+     * There was also a possible infinite loop when the number of
+     * elements was 1<<31, as |int i| was set to that, which is a
+     * negative number. However, it requires minimum input values:
+     *
+     * CRYPTO_secure_malloc_init((size_t)1<<34, (size_t)1<<4);
+     *
+     * Which really only works on 64-bit systems, and even then the
+     * code attempts to allocate 16 GB secure memory arena. Linux
+     * can deal with this better than other Unixy OS's (e.g. MacOS)
```

Comment:
For reference. The said limit seems to be available only on BSD and Linux. MacOS, Solaris, HP-UX, Tru64, IRIX, AIX don't have it. In the line-up MacOS is special in a way that it doesn't really limit amount of virtual address space, because it allocates swap files dynamically. In a sense you are limited by free space on system disk and not by fixed amount of swap space.